source/slang/slang-ir-specialize.cpp

```diff
@@ -2440,44 +2474,57 @@ struct SpecializationContext
         }
         if (elementCount == 0)
         {
-            auto resultValuePack = builder.emitMakeValuePack(0, (IRInst*const*)nullptr);
-            expandInst->replaceUsesWith(resultValuePack);
+            auto resultPack = makeSpecializedPack(builder, expandInst->getDataType(), elements.getArrayView());
+            expandInst->replaceUsesWith(resultPack);
             expandInst->removeAndDeallocate();
-            addUsersToWorkList(resultValuePack);
+            addUsersToWorkList(resultPack);
             return true;
         }
+        
+        bool isMultiBlock = as<IRYield>(expandInst->getFirstBlock()->getTerminator()) == nullptr;
 
         for (UInt i = 0; i < elementCount; i++)
         {
             IRCloneEnv cloneEnv;
-            IRBlock* firstBlock = nullptr;
             IRBuilder subBuilder = builder;
-            for (auto childBlock : expandInst->getBlocks())
+            IRBlock* mergeBlock = nullptr;
```

Comment:
Yong's note: here we extend the existing `specailizeExpand` logic so that it can work with `IRExpand` insts that about types in additional to just values. We need this because we just added a pass that turns all `expand` types into `IRExpand` insts as well.